File: app/src/main/java/com/mxt/anitrend/adapter/recycler/index/StudioAdapter.java
package com.mxt.anitrend.adapter.recycler.index;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Filter;

import androidx.annotation.NonNull;

import com.bumptech.glide.Glide;
import com.mxt.anitrend.R;
import com.mxt.anitrend.base.custom.recycler.RecyclerViewAdapter;
import com.mxt.anitrend.base.custom.recycler.RecyclerViewHolder;
import com.mxt.anitrend.databinding.AdapterStudioBinding;
import com.mxt.anitrend.model.entity.base.StudioBase;
import com.mxt.anitrend.util.CompatUtil;

import butterknife.OnClick;

/**
 * Created by max on 2017/12/20.
 */

public class StudioAdapter extends RecyclerViewAdapter<StudioBase> {

    public StudioAdapter(Context context) {
        super(context);
    }

    @NonNull
    @Override
    public RecyclerViewHolder<StudioBase> onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new StudioViewHolder(AdapterStudioBinding.inflate(CompatUtil.INSTANCE.getLayoutInflater(parent.getContext()), parent, false));
    }

    @Override
    public Filter getFilter() {
        return null;
    }

    protected class StudioViewHolder extends RecyclerViewHolder<StudioBase> {

        private AdapterStudioBinding binding;

        /**
         * Default constructor which includes binding with butter knife
         *
         * @param binding
         */
        public StudioViewHolder(AdapterStudioBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        /**
         * Load image, text, buttons, etc. in this method from the given parameter
         * <br/>
         *
         * @param model Is the model at the current adapter position
         */
        @Override
        public void onBindViewHolder(StudioBase model) {
            binding.setModel(model);
            binding.executePendingBindings();
        }

        /**
         * If any image views are used within the view holder, clear any pending async img requests
         * by using Glide.clear(ImageView) or Glide.with(context).clear(view) if using Glide v4.0
         * <br/>
         *
         * @see Glide
         */
        @Override
        public void onViewRecycled() {
            binding.unbind();
        }

        @Override @OnClick(R.id.container)
        public void onClick(View v) {
            performClick(clickListener, data, v);
        }

        @Override
        public boolean onLongClick(View v) {
            return performLongClick(clickListener, data, v);
        }
    }
}

File: app/src/main/java/com/mxt/anitrend/adapter/recycler/index/StudioAdapter.kt
package com.mxt.anitrend.adapter.recycler.index

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.Filter
import com.mxt.anitrend.R
import com.mxt.anitrend.base.custom.recycler.RecyclerViewAdapter
import com.mxt.anitrend.base.custom.recycler.RecyclerViewHolder
import com.mxt.anitrend.databinding.AdapterStudioBinding
import com.mxt.anitrend.extension.getLayoutInflater
import com.mxt.anitrend.model.entity.base.StudioBase

/**
 * Created by max on 2017/12/20.
 */
class StudioAdapter(
    context: Context,
) : RecyclerViewAdapter<StudioBase>(context) {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int,
    ): RecyclerViewHolder<StudioBase> = StudioViewHolder(
        AdapterStudioBinding.inflate(parent.context.getLayoutInflater(), parent, false),
    )

    override fun getFilter(): Filter? = null

    inner class StudioViewHolder(
        private val binding: AdapterStudioBinding,
    ) : RecyclerViewHolder<StudioBase>(binding.root) {
        init {
            bindClickListeners(R.id.container)
        }

        override fun onBindViewHolder(model: StudioBase) {
            binding.studioName.text = model.name
            binding.studioFavourite.visibility = if (model.isFavourite) View.VISIBLE else View.GONE
        }

        override fun onViewRecycled() = Unit

        override fun onClick(v: View) {
            performClick(clickListener, data, v)
        }

        override fun onLongClick(v: View): Boolean = performLongClick(clickListener, data, v)
    }
}
